blockchains/erc20/lib/timestamps_cache.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+const DATA_MISSING = -1;
+
+class TimestampsCache {
+  constructor() {
+    this.timestampStore = {};
+  }
+
+  saveTimestampInStore(blockNumber, timestamp) {
+    this.timestampStore[blockNumber] = timestamp;
+  }
+
+  getTimestampFromStore(blockNumber) {
+    if (Object.prototype.hasOwnProperty.call(this.timestampStore, blockNumber)) {
+      return this.timestampStore[blockNumber];
+    }
+
+    return DATA_MISSING;
+  }
+
+  async getTimestampFromNode(web3, blockNumber) {
+    console.log('Getting timestamp for ', blockNumber);
```

Comment:
Shouldn't we use the already defined `logger`

Reply:
I used it for debugging, removed now.